ui/src/app/nav.tsx

```diff
@@ -24,12 +33,20 @@ export function Aside({ isOpen, setIsOpen }: { isOpen: boolean, setIsOpen: Dispa
                         setCSRPEMString(e.target.result.toString());
                     }
                 }
-        };
-        reader.readAsText(file);
+            };
+            reader.readAsText(file);
+        }
+    };
+    const CSRValidationElement = function () {
+        try {
+            extractCSR(CSRPEMString)
+        } catch (e) {
+            return <div><i className="p-icon--error"></i>Invalid CSR</div>
```

Comment:
The tool currently does not tolerate a newline after the CSR, we should be able to handle this as this will be the case for any csr generated with openssl.

Example:

```
openssl req -newkey rsa:2048 -keyout PRIVATEKEY.key -out MYCSR.csr
```

Reply:
done, we just trim whitespace from all inputs.